### PR TITLE
yaml configuration

### LINKED ISSUE #63: Use YML for configuration file
@alexdavid suggested that YML is a good superset of JSON. I agree. Looking at the current config file format for Mycha, I don't see a strong reason to stick with coffee script. What do we think about using YML here @charlierudolph @alexdavid?

### PR DESCRIPTION
**Pull request by charlierudolph**
@alexdavid @kevgo 

Probably best to look at the diff without whitespace.

resolves #63

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,26 +35,23 @@ and provides a utility that gets tests up and running quickly
...
-    module.exports =
+    # Environment variables to add to process.env when running mocha
+    # mochaEnv:
 
-      # Environment variables to add to process.env when running mocha
-      mochaEnv: {}
+    # Default options to pass to mocha (can be overriden by command line options)
+    mochaOptions:
+      colors: true
+      compilers: 'coffee:coffee-script/register'
```
Comment (kevgo) [inline README.md]: I think you don't need to put strings in quotes in YML

File: README.md [hunk 2]
```diff
@@ -35,26 +35,23 @@ and provides a utility that gets tests up and running quickly
...
+      reporter: 'dot'
```
Comment (kevgo) [inline README.md]: same here

File: src/config_loader.coffee [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 path = require 'path'
+yaml = require 'js-yaml'
+
+
+parseConfig = (contents, done) ->
+  try
+    config = yaml.safeLoad contents
+  catch err
+    return done err
+  done null, config
```
Comment (kevgo) [inline src/config_loader.coffee]: This can go into the try block, then you can get rid of the `return` on line 10

File: src/config_loader.coffee [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
 load = (cwd, done) ->
-  configPath = path.join cwd, 'mycha.coffee'
+  configPath = path.join cwd, 'mycha.yml'
   fs.exists configPath, (exists) ->
-    config = {}
-    if exists
-      require 'coffee-script/register'
-      config = require configPath
-    done null, config
+    unless exists then return done null, {}
```
Comment (kevgo) [inline src/config_loader.coffee]: This reads a bit clunky. What do you think about:

```coffeescript
exists or return done null, {}

**Comment by kevgo** [comment]
Super awesome change!

File: src/config_loader.coffee (hunk 3 above)
Comment (charlierudolph) [inline src/config_loader.coffee]: If you put the done in the try, then it has the possibly of catching errors thrown in done, which isn't something I want to do

Comment (kevgo) [inline src/config_loader.coffee]: True.

File: README.md (hunk 1 above)
Comment (charlierudolph) [inline README.md]: removed quotes from all strings except testFilePattern as `*` has a special meaning in yaml.

File: README.md (hunk 2 above)
Comment (charlierudolph) [inline README.md]: removed

File: src/config_loader.coffee (hunk 4 above)
Comment (charlierudolph) [inline src/config_loader.coffee]: Coffeescript doesn't appear to like that syntax
```
> mycha@1.0.0 build /Users/charlesrudolph/github/mycha
> find ./src -name '*.coffee' -not -name '*_spec.coffee' | xargs coffee --compile --output lib/

/Users/charlesrudolph/github/mycha/src/config_loader.coffee:17:35: error: unexpected newline
    exists or return done null, {}
                                  ^
```

Comment (kevgo) [inline src/config_loader.coffee]: Argh, too bad!

**Comment by kevgo** [comment]
:ship:

**Pull request closed without merging**